src/container-comparison/CompareContainersWidget.tsx

```diff
@@ -160,6 +194,18 @@ export const CompareContainersWidget = ({ upstreamBlockId, downstreamBlockId }:
     parent: [],
   });
 
+  const { data } = useCourseContainerChildren(downstreamBlockId, true);
+  let showLocalUpdateAlert = false;
+  let localUpdateAlertBlockName = '';
+
+  // Show this alert if the only change is a local override to a text component
+  if (!isReadyToSyncIndividually && data?.upstreamReadyToSyncChildrenInfo.length === 1
+        && data.upstreamReadyToSyncChildrenInfo[0].isModified
+        && data.upstreamReadyToSyncChildrenInfo[0].blockType === 'html') {
+    showLocalUpdateAlert = true;
+    localUpdateAlertBlockName = data.upstreamReadyToSyncChildrenInfo[0].name;
+  }
+
```

Comment:
This section can be moved inside `CompareContainersWidgetInner` component and duplicate calls to `useCourseContainerChildren` api  can be avoided. As we only want to fetch `getUpstreamInfo` at top level in diff preview, we can use `parent.length==0` like so:

```tsx
const { data, isError, error } = useCourseContainerChildren(downstreamBlockId, parent.length === 0);
```